pdp/handlers.go

```diff
@@ -909,15 +914,213 @@ func (p *PDPService) handleAddRootToProofSet(w http.ResponseWriter, r *http.Requ
+		return
+	}
+
+	if proofSetService != serviceLabel {
+		// Same response as not found to avoid leaking information
+		http.Error(w, "Proof set not found", http.StatusNotFound)
+		return
+	}
+
+	// Step 4: Query pdp_proofset_root_adds for this transaction
+	type RootAddInfo struct {
+		Root            string `db:"root"`
+		AddMessageIndex int    `db:"add_message_index"`
+		Subroot         string `db:"subroot"`
+		SubrootOffset   int64  `db:"subroot_offset"`
+		SubrootSize     int64  `db:"subroot_size"`
+		AddMessageOK    *bool  `db:"add_message_ok"`
+		RootsAdded      bool   `db:"roots_added"`
+	}
+
+	var rootAdds []RootAddInfo
+	err = p.db.Select(ctx, &rootAdds, `
+		SELECT root, add_message_index, subroot, subroot_offset,
+		       subroot_size, add_message_ok, roots_added
+		FROM pdp_proofset_root_adds
+		WHERE proofset = $1 AND add_message_hash = $2
+		ORDER BY add_message_index, subroot_offset
+	`, proofSetID, txHash)
+	if err != nil {
+		http.Error(w, "Failed to query root additions: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if len(rootAdds) == 0 {
+		http.Error(w, "Root addition not found for given transaction", http.StatusNotFound)
+		return
+	}
+
+	// Step 5: Get transaction status from message_waits_eth
+	var txStatus string
+	err = p.db.QueryRow(ctx, `
+		SELECT tx_status FROM message_waits_eth WHERE signed_tx_hash = $1
+	`, txHash).Scan(&txStatus)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			http.Error(w, "Transaction status not found", http.StatusNotFound)
+			return
+		}
+		http.Error(w, "Failed to query transaction status: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// Step 6: If transaction is confirmed and successful, get assigned root IDs
+	var confirmedRootIds []uint64
+	if txStatus == "confirmed" && len(rootAdds) > 0 && rootAdds[0].AddMessageOK != nil && *rootAdds[0].AddMessageOK {
+		// Get unique roots from the add transaction
+		uniqueRoots := make(map[string]bool)
+		for _, ra := range rootAdds {
+			uniqueRoots[ra.Root] = true
+		}
+
+		// Query pdp_proofset_roots for confirmed roots with their IDs
+		rootCids := make([]string, 0, len(uniqueRoots))
+		for root := range uniqueRoots {
+			rootCids = append(rootCids, root)
+		}
+
+		type ConfirmedRoot struct {
+			RootID uint64 `db:"root_id"`
+			Root   string `db:"root"`
+		}
+
+		var confirmedRoots []ConfirmedRoot
+		err = p.db.Select(ctx, &confirmedRoots, `
+			SELECT DISTINCT root_id, root
+			FROM pdp_proofset_roots
+			WHERE proofset = $1 AND root = ANY($2)
+			ORDER BY root_id
+		`, proofSetID, rootCids)
+		if err != nil {
+			log.Warnf("Failed to query confirmed roots: %v", err)
+			// Don't fail the request, just log the warning
+		} else {
+			// Extract just the root IDs
+			for _, cr := range confirmedRoots {
+				confirmedRootIds = append(confirmedRootIds, cr.RootID)
+			}
+		}
+	}
+
+	// Step 7: Build and send response
+	// Count unique roots for response
+	uniqueRootMap := make(map[string]bool)
```

Comment:
Am I missing something or is this just AI being AI?  Declare this before the if statement so you don't have to do this twice.

Reply:
Ok this makes sense you might not eter the if above but still need this.  But can you do this above the if statement?  Its the exact map constructed from the exact same data in both cases right?  I dont understand why we would do this twice

Reply:
Im talking about 1043 to 1046not confirmedRoots

Reply:
yeah, sorry, I get you now, I'm just being blinkered on the steps here; if we just do this calc outside of the first `if` then we're all good.